QEfficient/peft/auto.py

```diff
@@ -0,0 +1,436 @@
+        if kwargs.get("use_cache") is False:
+            warnings.warn("Overriding to use_cache=True")
+        kwargs["use_cache"] = True
+        obj = cls._from_pretrained(pretrained_name_or_path, *args, **kwargs)
+        obj.load_adapter(pretrained_name_or_path, obj.active_adapter)
+        return obj
+
+    def transform(self, **kwargs):
+        # Base class
+        for transform in self._pytorch_transforms:
+            self.model, transformed = transform.apply(self.model)
+        logger.info("Pytorch transforms applied")
+
+    def _export(
+        self,
+        sample_inputs: Dict[str, torch.Tensor],
+        input_names: List[str],
+        output_names: List[str],
+        dynamic_axes: Dict[str, Dict[int, str]],
+        export_kwargs: Dict[str, any] = {},
+        onnx_transform_kwargs: Dict[str, any] = {},
+        export_dir: Optional[str] = None,
+    ) -> str:
+        # Base class
+        export_dir = Path(export_dir or (QEFF_HOME / self.model_name))
+        export_dir = export_dir.with_name(export_dir.name + "-" + self.model_hash)
+        onnx_path = export_dir / f"{self.model_name}.onnx"
+        if onnx_path.is_file():
+            self.onnx_path = onnx_path
+            return onnx_path
+
+        tmp_onnx_dir = export_dir / "onnx_tmp"
+        tmp_onnx_path = tmp_onnx_dir / f"{self.model_name}.onnx"
+        tmp_onnx_dir.mkdir(parents=True, exist_ok=True)
+
+        try:
+            torch.onnx.export(
+                self.model,
+                (sample_inputs,),
+                str(tmp_onnx_path),
+                input_names=input_names,
+                output_names=output_names,
+                dynamic_axes=dynamic_axes,
+                opset_version=13,
+                **export_kwargs,
+            )
+            logger.info("Pytorch export successful")
+
+            model = onnx.load(tmp_onnx_path, load_external_data=False)
+            onnx_transform_kwargs = {
+                "onnx_base_dir": str(tmp_onnx_dir),
+                "model_name": self.model_name,
+                **onnx_transform_kwargs,
+            }
+            for transform in self._onnx_transforms:
+                model, transformed = transform.apply(model, **onnx_transform_kwargs)
+            model.metadata_props.append(
+                onnx.StringStringEntryProto(key="qeff_transforms", value=",".join(self.transform_names()))
+            )
+            logger.info("ONNX transforms applied")
+
+            onnx.save(model, onnx_path)
+            logger.info("Transformed onnx saved")
+
+        except Exception as e:
+            logger.error(f"ONNX export failed: {e}")
+            raise e
+
+        finally:
+            shutil.rmtree(tmp_onnx_dir, ignore_errors=True)
+
+        self.onnx_path = onnx_path
+        return onnx_path
+
+    def export(self, export_dir: Optional[str] = None) -> str:
+        kv_cache_shape = get_padding_shape_from_config(self.model.config, 1, 32)
```

Comment:
Why are we fixing `batch_size` and `seq_len`?

Reply:
We can have it in the constants, but it doesn't matter because the resulting onnx is going to be dynamic anyways. I'll move it to constants still.

Reply:
Done